lib/components/ui/file-input.tsx

```diff
@@ -1,19 +1,42 @@
 import * as React from 'react';
+import { cn } from '@/utils';
 import ImageBoxFull from '@/assets/img_box_fill.svg?react';
+import CloseIcon from '@/assets/close.svg?react';
 import { Input } from './input';
 
 export interface FileInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
   onAdd: (file: File) => void;
+  onFileRemove?: () => void;
   error?: string | boolean;
 }
 
 const FileInput = React.forwardRef<HTMLInputElement, FileInputProps>(
-  ({ className, onAdd, error, ...props }, ref) => {
+  ({ className, onAdd, onFileRemove, error, ...props }, ref) => {
     const [fileName, setFileName] = React.useState('');
     const handleInputChange = (files: FileList) => {
       setFileName(files[0].name);
       onAdd(files[0]);
     };
+
+    const getRightIconGroup = (onFileRemove?: () => void) => (
```

Comment:
onFileRemove можно не передавать, так как он глобально доступен в компоненте. В целом можно вообще эту верстку оставить прямо в пропсе rightIcon. А то компонент маленький, на 50 строчек, поэтому так даже легче читать код будет, как мне кажется.

```suggestion
    const renderRightIconGroup = () => (
```